boards/metro_m0/src/lib.rs

```diff
@@ -230,23 +232,24 @@ pub fn uart<F: Into<Hertz>>(
 }
 
 #[cfg(feature = "usb")]
-pub fn usb_bus(
+pub fn usb_allocator(
     usb: pac::USB,
     clocks: &mut GenericClockController,
     pm: &mut pac::PM,
-    dm: gpio::Pa24<Input<Floating>>,
-    dp: gpio::Pa25<Input<Floating>>,
+    usb_dm: gpio::Pa24<Input<Floating>>,
+    usb_dp: gpio::Pa25<Input<Floating>>,
     port: &mut Port,
-) -> UsbBusWrapper<UsbBus> {
-    let gclk0 = clocks.gclk0();
-    dbgprint!("making usb clock");
-    let usb_clock = &clocks.usb(&gclk0).unwrap();
-    dbgprint!("got clock");
-    UsbBusWrapper::new(UsbBus::new(
-        usb_clock,
-        pm,
-        dm.into_function(port),
-        dp.into_function(port),
-        usb,
-    ))
+    ) -> UsbBusAllocator<UsbBus> {
+    clocks.configure_gclk_divider_and_source(GEN_A::GCLK2, 1, SRC_A::DFLL48M, false);
+    let usb_gclk = clocks.get_gclk(GEN_A::GCLK2).unwrap();
+    let usb_clock = &clocks.usb(&usb_gclk).unwrap();
```

Comment:
Do you think this should configure gclk2 or take the usb clock as an argument instead? An app on top of this board crate could want to use gclk2 for something else?

Reply:
Yeah, taking the USB clock probably makes sense. This was one of the major copy-paste points where I wasn't really sure about the right approach.

Reply:
:+1: lets switch to that

Reply:
I'm not totally sure what to do for this. I poked through the pac and hal for a bit and didn't really see anything obvious to use here. I skimmed the [data sheet](http://ww1.microchip.com/downloads/en/Appnotes/Atmel-42261-SAM-D21-USB_Application-Note_AT06475.pdf) a bit and noticed on page 11 it recommends using gclk0 as the default for USB unless you specifically need something else, so I'll switch it to gclk0.

Reply:
That sounds good, `GenericClockController` already initialises gclk0 to 48MHz and the example is using the external oscillator :+1: